### PR TITLE
Video subjects

### PR DESCRIPTION
**Pull request by chelseatroy**
# What this change should allow you to do:

ON a swipe layout project, you can now include video subjects! In staging, you can test this on the video subject workflow of the "All Mobile Workflows" project.

# Review Checklist

- [x] Does it work in Android and iOS?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Are tests passing?

# My goals for this PR:

### CODE REVIEW

File: src/components/FullScreenMedia.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        function displayWithRequisiteComponent(uri, handlePress) {
+            if (uri.slice(uri.length - 4).match('.mp4')) {
+                return <Video
+                    source={{uri: uri}}
+                    style={{
+                        width: Dimensions.get('window').width,
+                        height: Dimensions.get('window').height
+                    }}
+                    controls={true}
+                    repeat={true}
```
Comment (mcbouslog) [inline src/components/FullScreenMedia.js:27]: The video behavior on the Zooniverse web app is for video not to repeat. I think there are accessibility reasons, but I'm having difficulty finding a source. At minimum for consistency I lean towards not repeating, but don't have a strong opinion, defer to you.

File: src/components/classifier/SwipeableSubject.js [hunk 2]
```diff
@@ -50,6 +51,25 @@ class SwipeCardSubjectsView extends Component {
...
         const { imageUris, hasMultipleSubjects, onDisplayImageChange } = this.props;
         const imagesAreLoaded = !R.isEmpty(imageUris);
+
+        function displayInRequisiteComponent(uri) {
+            if (uri.slice(uri.length - 4).match('.mp4')) {
+                return <Video
+                    source={{ uri: uri }}
+                    style={{ width: pagerDimensions.width, height: pagerDimensions.height }}
+                    controls={true}
+                    repeat={true}
```
Comment (mcbouslog) [inline src/components/classifier/SwipeableSubject.js:61]: See previous comment regarding repeating.

**Review by mcbouslog — APPROVED** [review]
This is great! Code looks good, works well on simulated iOS and Android devices.

One comment (related to the two instances the video component is used) related to repeating the video.

Not related to these changes, but maybe these changes cause it to continue to happen - in the ClassifierContainer.js, `onFeedbackViewLayout`, adding `useNativeDriver: false` [after line 50](https://github.com/zooniverse/mobile/blob/master/src/components/classifier/ClassifierContainer.js#L50) prevents a warning that otherwise continually logs.

**Comment by mcbouslog** [comment]
I think this was noted elsewhere (Slack?), but some button text alignment may have changed:
Screen Shot 2022-01-14 at 3 36 18 PM
I can open a dedicated issue if that'd be helpful?

**Comment by mcbouslog** [comment]
Possibly another separate issue to be opened, I'm getting the following error after swiping a video subject now:
Screen Shot 2022-01-14 at 5 08 56 PM
I think it may have to do with my subject queue, but not sure. I can open an issue and attempt to investigate more, but want to mention here if already aware of?

File: src/components/FullScreenMedia.js (hunk 1 above)
Comment (chelseatroy) [reply src/components/FullScreenMedia.js:27]: Totally. The reason for repeating in this context is that this library offers minimal control over playback for users, and these videos are short enough that it seems likely that the volunteer will not catch the action the first time. So I'd like to leave this for mobile for now.

**Comment by mcbouslog** [comment]
> Possibly another separate issue to be opened, I'm getting the following error after swiping a video subject on Android: Screen Shot 2022-01-14 at 5 08 56 PM I think it may have to do with my subject queue, but not sure. I can open an issue and attempt to investigate more, but want to mention here if already aware of?

I am unable to recreate this issue, please disregard.

**Comment by mcbouslog** [comment]
Closes #290

**Pull request merged**